### PR TITLE
Allocate space in header for floating-point sample interval + offset,…

### PR DESCRIPTION
**Pull request by da-wad**
… and test with ZGY

### CODE REVIEW

File: docs/file-specification.md [hunk 1]
```diff
@@ -48,7 +48,9 @@ Byte encoding is little-endian.
 |72-75 |uint32 |Encoded version number
 |76-79 |uint32 |Encoded source format 0=SEG-Y, 10=ZGY, 20=numpy
 |80-83 |uint32 |Encoded header-detection method ***
-|84-959 |---  | --- Unused ---
+|84-92   |float64  | Minimum sample time/depth*****
```
Comment (ThomasRyschawyEquinor) [inline docs/file-specification.md:51]: Here you want to use 91: 84-91

File: docs/file-specification.md [hunk 2]
```diff
@@ -48,7 +48,9 @@ Byte encoding is little-endian.
...
+|92-100  |float64  | Sample interval (μs/m)*****
```
Comment (ThomasRyschawyEquinor) [inline docs/file-specification.md:52]: same here

File: seismic_zfp/utils.py [hunk 3]
```diff
@@ -100,9 +100,18 @@ def coord_to_index(coord, coords, include_stop=False):
 
 
 def gen_coord_list(start, step, count):
+    print(np.arange(start, start + step*count, step))
```
Comment (ThomasRyschawyEquinor) [inline seismic_zfp/utils.py:103]: remove this line

File: tests/test_compress.py [hunk 4]
```diff
@@ -51,6 +51,8 @@
 ZGY_FILE_32 = 'test_data/zgy/small-32bit.zgy'
 ZGY_FILE_16 = 'test_data/zgy/small-16bit.zgy'
 ZGY_FILE_8 = 'test_data/zgy/small-8bit.zgy'
+ZGY_FILE_FLOAT_SR = 'test_data/zgy/small-float-samplerate.zgy'
+SGY0_FILE_FLOAT_SR = 'test_data/zgy/small-float-samplerate-rev0.sgy'
```
Comment (ThomasRyschawyEquinor) [inline tests/test_compress.py:55]: typo SGY0_ should be SGY_?

File: docs/file-specification.md (hunk 1 above)
Comment (da-wad) [reply docs/file-specification.md:51]: Yes, looks like this is the interval format here...

File: docs/file-specification.md (hunk 2 above)
Comment (da-wad) [reply docs/file-specification.md:52]: Yes, looks like this is the interval format here...

File: seismic_zfp/utils.py (hunk 3 above)
Comment (da-wad) [reply seismic_zfp/utils.py:103]: Yes, good to get rid of this!

File: seismic_zfp/conversion_utils.py [hunk 5]
```diff
@@ -40,6 +41,10 @@ def make_header_seismic_file(seismicfile, bits_per_voxel, blockshape, geom, head
             segy_file_header = f.read(SEGY_FILE_HEADER_BYTES)
             buffer[DISK_BLOCK_BYTES:DISK_BLOCK_BYTES + SEGY_FILE_HEADER_BYTES] = segy_file_header
 
+    elif seismicfile.filetype == Filetype.ZGY:
+        buffer[84:92] = double_to_bytes(seismicfile.samples[0])
+        buffer[92:100] = double_to_bytes(seismicfile.zinc*1000)
```
[the remark this answers is not in the document]
Comment (da-wad) [reply seismic_zfp/conversion_utils.py:46]: It's double-precision floating point, unless you're going off the end of the range of the exponent you'll only lose the 13th or so decimal digit in precision...

File: tests/test_compress.py (hunk 4 above)
Comment (ThomasRyschawyEquinor) [reply tests/test_compress.py]: Or maybe remove the line since it is not being used

Comment (ThomasRyschawyEquinor) [reply tests/test_compress.py]: I now actually wonder if this fix is only for ZGY files and not SGY, right?

**Review by ThomasRyschawyEquinor — COMMENTED** [review]
Good job! See some comments and questions

**Review by ThomasRyschawyEquinor — APPROVED** [review]
Nice work. Looks good to me

File: tests/test_compress.py (hunk 4 above)
Comment (da-wad) [reply tests/test_compress.py]: It is only for ZGY files, yes. SEG-Y doesn't support this until rev.2 ... and there are no opensource rev.2 reading/writing libraries yet ¯\_(ツ)_/¯

**Pull request merged**